Shared/BrainCloudAppStore.mm

```diff
@@ -0,0 +1,167 @@
+#include "BrainCloudCallback.hh"
+
+#import "BrainCloudAppStore.hh"
+#import "BrainCloudClient.hh"
+
+
+@interface BrainCloudAppStore ()
+{
+    BrainCloud::BrainCloudClient *_client;
+}
+@end
+
+@implementation BrainCloudAppStore
+
+- (instancetype) init: (BrainCloudClient*) client
+{
+    self = [super init];
+    
+    if(self) {
+        _client = (BrainCloud::BrainCloudClient *)[client getInternalClient];
+    }
+    
+    return self;
+}
+
+/**
+ * Verifies that purchase was properly made at the store
+ *
+ * Service Name - AppStore
+ * Service Operation - VERIFY_PURCHASE
+ *
+ * @param storeId the store platform. Valid: itunes, facebook, appworld, steam, windows, windowsPhone, googlePlay
+ * @param receiptData the specific store data required
+ * @param completionBlock Block to call on return of successful server response
+ * @param errorCompletionBlock Block to call on return of unsuccessful server response
+ * @param cbObject User object sent to the completion blocks
+ */
+-(void)verifyPurchase:(NSString *)storeId
+          receiptData:(NSString *)receiptData
+      completionBlock:(BCCompletionBlock)cb
+ errorCompletionBlock:(BCErrorCompletionBlock)ecb
+             cbObject:(BCCallbackObject)cbObject
+{
+    _client->getAppStoreService()->verifyPurchase(
+         [storeId UTF8String], [receiptData UTF8String], new BrainCloudCallback(cb, ecb, cbObject));
+}
+
+/**
+ * Returns the eligible promotions for the player
+ *
+ * Service Name - AppStore
+ * Service Operation - ELIGIBLE_PROMOTIONS
+ *
+ * @param completionBlock Block to call on return of successful server response
+ * @param errorCompletionBlock Block to call on return of unsuccessful server response
+ * @param cbObject User object sent to the completion blocks
+ */
+- (void)getEligiblePromotions:(BCCompletionBlock)cb
+         errorCompletionBlock:(BCErrorCompletionBlock)ecb
+                     cbObject:(BCCallbackObject)cbObject
+{
+    _client->getAppStoreService()->getEligiblePromotions(new BrainCloudCallback(cb, ecb, cbObject));
+}
+
+/**
+ * gets the active sales inventory for the passed in currencytype
+ *
+ * Service Name - AppStore
+ * Service Operation - GET_INVENTORY
+ *
+ * @param storeId the store platform. Valid: itunes, facebook, appworld, steam, windows, windowsPhone, googlePlay
+ * @param userCurrency the currency type to retrieve the inventory for
+ * @param category product category
+ * @param completionBlock Block to call on return of successful server response
+ * @param errorCompletionBlock Block to call on return of unsuccessful server response
+ * @param cbObject User object sent to the completion blocks
+ */
+- (void)getSalesInventory:(NSString *)storeId
+             userCurrency:(NSString *)userCurrency
+          completionBlock:(BCCompletionBlock)cb
+     errorCompletionBlock:(BCErrorCompletionBlock)ecb
+                 cbObject:(BCCallbackObject)cbObject
+{
+    _client->getAppStoreService()->getSalesInventoryByCategory(
```

Comment:
Hum not a big deal, but there's a `getSalesInventory` in the C++ API also. Could have just called that.
Can you make that small change?